### PR TITLE
add attribute for max frame size

### LINKED ISSUE #303: [Enhancement]add configuration attribute query for max frame size
As the request for enabling max frame size for ffmpeg-vaapi, a query for the support of driver is need. 
Otherwise the feature can only be enabled by hard code, and won't be accept by community easily.

https://patchwork.ffmpeg.org/patch/12932/

### PR DESCRIPTION
**Pull request by XinfengZhang**
Fixes #303 
Signed-off-by: Carl Zhang <carl.zhang@intel.com>

### CODE REVIEW

File: va/va.h [hunk 1]
```diff
@@ -941,6 +946,21 @@ typedef union _VAConfigAttribValDecJPEG {
 #define VA_ENC_SLICE_STRUCTURE_ARBITRARY_ROWS           0x00000010
 /**@}*/
 
+/** \brief Attribute value for VAConfigAttribMaxFrameSize */
+typedef union _VAConfigAttribValMaxFrameSize{
+    struct{
+        /** \brief support max frame size 
+          * if max_frame_zie == 1, VAEncMiscParameterTypeMaxFrameSize/VAEncMiscParameterBufferMaxFrameSize
```
Comment (uartie) [inline va/va.h]: typo: max_frame_zie

File: va/va.h [hunk 2]
```diff
@@ -941,6 +946,21 @@ typedef union _VAConfigAttribValDecJPEG {
...
+          * could be used to set the frame size, if multiple_pass also equal 1, VAEncMiscParameterTypeMultiPassFrameSize
+          * VAEncMiscParameterBufferMultiPassFrameSize could be used to set frame size and pass information 
+          */ 
+        uint32_t max_frame_size : 1;
+        /** \brief multiple_pass support */
+        uint32_t multiple_pass  : 1;
+    }bits;
+    uint32_t value;
+}VAConfigAttribValMaxFrameSize;
```
Comment (uartie) [inline va/va.h]: formatting does not follow other code.  E.g. need spaces before/after braces.

**Comment by XinfengZhang** [comment]
hi @xhaihao , could you review it

File: va/va.h [hunk 3]
```diff
@@ -941,6 +946,21 @@ typedef union _VAConfigAttribValDecJPEG {
...
+typedef union _VAConfigAttribValMaxFrameSize {
+    struct {
+        /** \brief support max frame size 
+          * if max_frame_size == 1, VAEncMiscParameterTypeMaxFrameSize/VAEncMiscParameterBufferMaxFrameSize
+          * could be used to set the frame size, if multiple_pass also equal 1, VAEncMiscParameterTypeMultiPassFrameSize
+          * VAEncMiscParameterBufferMultiPassFrameSize could be used to set frame size and pass information 
+          */ 
+        uint32_t max_frame_size : 1;
+        /** \brief multiple_pass support */
+        uint32_t multiple_pass  : 1;
```
Comment (xhaihao) [inline va/va.h]: It would be better to make sure the reserved bits are 0,

**Pull request merged**